modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/endpoint/IngestRestService.java

```diff
@@ -838,7 +847,8 @@ public Response addMediaPackage(@Context HttpServletRequest request, @PathParam(
               } catch (java.net.URISyntaxException e) {
                 return badRequest(String.format("Invalid URI %s for media", value), e);
               }
-              ingestService.addTrack(mediaUrl, flavor, mp);
+              ingestService.addTrack(mediaUrl, flavor, tags.toArray(new String[0]), mp);
```

Comment:
Ok, bare with me here.  We want to be able to tag, and flavor some combination of media files, and dc catalogs.  To do this we are forced to rely on the *ordering* of the parameters in the request, right?

In that case shouldn't we be clearing `flavor` as well, since otherwise it's going to be applied to anything *else* we might ingest as well?  c.f: a request with parameter ordering `flavor`, `tag`, `t1`, `t2` is going to apply the flavor and tag to `t1`, but only the flavor to `t2`.

Reply:
Hood point. But removing functionality from the ingest API is dangerous, so I rather not do that.
I could try and make this consistent by only clearing this if new tags are set after a media file:

```
flavor=a
tag=b
tag=c
track_a (with flavor a and tags b and c)
track_b (with flavor a and tags b and c)
```

and

```
flavor=a
tag=b
tag=c
track_a (with flavor a and tags b and c)
flavor=x
tag=y
track_b (with flavor x and tag y)
```
Or do you think it is save to disallow a second track without dedicated flavor?

Reply:
I think it's likely safe to disallow tracks without flavours. My read of the rest docs says that you need to ingest (flavor, track) pairs already.

I'm also struggling to imagine a legit usecase for setting a single flavour and ingesting multiple files. Concatenation after hard clips/crashes on the CA?

Reply:
As discussed at the technical meeting, a warning has now been added in 588a4df8a8f0ce63bed3cab0ae6916b41b9cf7ca